kroxylicious/src/main/java/io/kroxylicious/proxy/config/micrometer/MicrometerConfiguration.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.proxy.config.micrometer;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.binder.MeterBinder;
+
+import io.kroxylicious.proxy.classloader.ClassloaderUtils;
+
+public class MicrometerConfiguration {
+
+    private static final MicrometerConfiguration EMPTY = new MicrometerConfiguration(null, null, null);
+    private static final List<String> BINDER_PACKAGES = getBinderPackages();
+
+    private final List<Class<? extends MeterBinder>> binders;
+    private final Class<? extends MicrometerConfigurationHook> configurationHook;
+    private final List<Tag> commonTags;
+
+    private static List<String> getBinderPackages() {
+        String binderPackage = "io.micrometer.core.instrument.binder";
+        return List.of(binderPackage + ".jvm", binderPackage + ".system");
+    }
+
+    public MicrometerConfiguration(List<String> binders, String configurationHookClass, Map<String, String> commonTags) {
+        this.binders = configureBinders(binders);
+        this.configurationHook = getConfigurationHookClass(configurationHookClass);
+        this.commonTags = toTags(commonTags);
+    }
+
+    public static MicrometerConfiguration empty() {
+        return EMPTY;
+    }
+
+    private List<Tag> toTags(Map<String, String> commonTags) {
+        if (commonTags == null) {
+            return List.of();
+        }
+        return commonTags.entrySet().stream().map(entry -> Tag.of(entry.getKey(), entry.getValue())).collect(Collectors.toList());
+    }
+
+    private Class<? extends MicrometerConfigurationHook> getConfigurationHookClass(String configurationHookClass) {
```

Comment:
`Optional<Class<...>>` instead of a `null` return value.

Reply:
done. I also made the field `Optional` despite the general recommendation not to (because `Optional` is not Serializable).